### PR TITLE
995-click-to-copy-input

### PR DESCRIPTION
**Pull request by ScotianMan**
# Description 

Adds click to copy functionality to the secret share url created on the moments page. The back end is set up for i18n and the var is set for the front end but react is not playing nicely so for now the confirmation toast will be in english but the tool tip is international providing someone translates. I added a dependency for `react-toastify` but I couldnt find we had any toast implemented at all and using a tooltip didnt feel/look right. 

## Corresponding Issue

#995 

# Screenshots

https://screencast.com/t/k4vbGGmAVsYd

# Test Coverage

✅ however a feature test confirming the toast firing and clipboard copy could be added.

### CODE REVIEW

**Comment by ScotianMan** [comment]
Anyone know why there are controller specs in the helpers spec? Its causing a conflict, I can resolve it easily enough but should there be controller specs in a helper file on master?
/cc @julianguyen

File: client/app/components/Input/InputDefault.jsx [hunk 1]
```diff
@@ -79,6 +89,10 @@ export const InputDefault = (props: Props) => {
       }
       ref={myRef}
       aria-label={label}
+      onClick={(e: SyntheticEvent<HTMLInputElement>) => copyToClipBoard(e)}
```
[the remark this answers is not in the document]
Comment (julianguyen) [reply client/app/components/Input/InputDefault.jsx]: We should write an integration test for this in InputDefault.spec.jsx

**Comment by julianguyen** [comment]
As for the merge conflicts, I would just resolve them for now and you can open up a new issue if you'd like to tackle the separation of concerns issue you raised. Thanks!

File: client/app/components/Input/InputDefault.jsx [hunk 2]
```diff
@@ -21,6 +23,14 @@ export type Props = {
   hasError?: Function,
   myRef?: any,
   label?: string,
+  onClick?: Function,
+};
+
+const copyToClipBoard = (e: SyntheticEvent<HTMLInputElement>) => {
+  e.currentTarget.select();
+  document.execCommand('copy');
+  // TODO: Once i18n and react are playing nicely this can be changed
```
Comment (julianguyen) [inline client/app/components/Input/InputDefault.jsx]: Did you encounter issues as well?

File: client/app/components/Input/InputDefault.jsx [hunk 3]
```diff
@@ -60,7 +70,7 @@ export const InputDefault = (props: Props) => {
   } = props;
   if (!REQUIRES_DEFAULT.includes(type)) return null;
   return (
-    <input
+    <><input
```
Comment (julianguyen) [inline client/app/components/Input/InputDefault.jsx]: I'm wondering why `<>` was added.

File: app/helpers/moments_helper.rb [hunk 4]
```diff
@@ -15,8 +15,25 @@ def edit_moment_props
     edit_form_props(moment_form_inputs, moment_path(@moment))
   end
 
+  def secret_share_url_props(moment)
```
Comment (jm96441n) [inline app/helpers/moments_helper.rb]: why is this here just to call a private method? why not expose that method?

File: spec/helpers/moments_helper_spec.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
+describe MomentsHelper do
+  let(:user) { FactoryBot.create(:user) }
+  let(:moment) { FactoryBot.create(:moment, :with_secret_share, user: user) }
```
Comment (jm96441n) [inline spec/helpers/moments_helper_spec.rb]: do these need to be persisted? can we use `build` instead and stub the DB calls?

File: app/assets/stylesheets/secret_share/secret_share_content.scss [hunk 6]
```diff
@@ -5,3 +5,13 @@
   background: $white-80;
   @include dashboardSectionStyles();
 }
+
+.secretShare {
+  form {
```
[the remark this answers is not in the document]
Comment (ScotianMan) [reply app/assets/stylesheets/secret_share/secret_share_content.scss]: sure does, fixed 😜 That form was overriding everything on me for nearly 20 minutes and when I made the css finally listen to me I never looked back at fixing it. CSS and I have a love hate relationship 😂

File: client/package.json [hunk 7]
```diff
@@ -103,7 +104,7 @@
   },
   "jest": {
     "moduleNameMapper": {
-      "\\.scss$": "identity-obj-proxy"
+      "\\.(css|scss)$": "identity-obj-proxy"
```
[the remark this answers is not in the document]
Comment (ScotianMan) [reply client/package.json]: The toastify css was not being parsed by Jest, this was why :-)

File: client/app/components/Input/InputDefault.jsx (hunk 3 above)
Comment (ScotianMan) [reply client/app/components/Input/InputDefault.jsx]: so I was only returning one element. I changed it to a div

File: app/helpers/moments_helper.rb (hunk 4 above)
Comment (ScotianMan) [reply app/helpers/moments_helper.rb]: To protect this trivial logic and future logic being used to generate the props as well as to keep the style of the file in line with the other functions in this helper. Is there a reason not to? style or security?

Comment (jm96441n) [reply app/helpers/moments_helper.rb]: I'd say more in terms of OO design. Generally private methods are in place to hide implementation details that aren't necessary for the public interface of an object, but since the only thing this method does is call a private method (which returns a hash, so the internals of the method are still exposed) it doesn't necessarily need to be private. Also, in ruby, `private` methods shouldn't really be used as a form of security since they aren't really private as in other OO languages

**Comment by julianguyen** [comment]
Just checking in - any updates on this?

**Comment by julianguyen** [comment]
I'm going to finish this issue and merge it into `finish-click-to-copy-input`
